lib/hanami.rb

```diff
@@ -67,7 +63,23 @@ def self.shutdown
     application.shutdown
   end
 
+  def self.bundled?(gem_name)
+    @_mutex.synchronize do
+      @_bundled[gem_name] ||= begin
+        gem(gem_name)
+        true
+      rescue Gem::LoadError
+        false
+      end
+    end
+  end
+
   def self.bundler_groups
     [:plugins]
   end
+
+  require_relative "hanami/version"
+  require_relative "hanami/errors"
+  require_relative "hanami/extensions"
+  require_relative "hanami/application"
```

Comment:
@jodosha Is there a reason you keep these nested inside the `module Hanami` rather than just at the bottom of the file?

I think I'd prefer having them right at the bottom, because in reading this, having them nested seems to signify that that nesting is critical in some way, when it's actually not (I moved them to the bottom when playing around with this code and everything still worked).

Reply:
@timriley "extensions" depend on Hanami.bundled?, which is defined above in this file.

Reply:
@jodosha Right! So would we have just that one at the end of the file, and the rest at the top? That would give a clearer signal that "extensions" is the special case.